src/EditorFeatures/Core/Classification/Semantic/AbstractSemanticOrEmbeddedClassificationViewTaggerProvider.cs

```diff
@@ -20,111 +18,90 @@
 using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Text.Editor;
 using Microsoft.VisualStudio.Text.Tagging;
-using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.Classification
+namespace Microsoft.CodeAnalysis.Classification;
+
+/// <summary>
+/// This is the tagger we use for view classification scenarios.  It is used for classifying code
+/// in the editor.  We use a view tagger so that we can only classify what's in view, and not
+/// the whole file.
+/// </summary>
+internal abstract class AbstractSemanticOrEmbeddedClassificationViewTaggerProvider : AsynchronousViewportTaggerProvider<IClassificationTag>
```

Comment:
this is the most critical type being updated.  It is both very expensive (because it uses semantics), and it runs almost immediately after any edit (like typing, etc.).  So while typing, it was virtually guaranteed that after every keystroke, this would be reclassifying a very large extra region of the file, despite rarely being needed.  This causes tons of computation in the compiler layer binding and other semantic analysis.